src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBarracksTowerNew.java

```diff
@@ -11,51 +11,59 @@
 import org.jetbrains.annotations.NotNull;
 
 /**
- * Building class of the BarracksTower.
+ * Building class for the Barracks Tower.
  */
-public class BuildingBarracksTower extends AbstractBuildingGuards
+public class BuildingBarracksTowerNew extends AbstractBuildingGuardsNew
 {
-    /**
-     * Name description of the guard hat.
-     */
-    private static final String GUARD_TOWER = "BarracksTower";
-
-    /**
-     * Offence bonus related to this building.
-     */
-    private static final int OFFENCE_BONUS = 5;
+    ////// --------------------------- NBTConstants --------------------------- \\\\\\
+    private static final String TAG_POS = "pos";
+    ////// --------------------------- NBTConstants --------------------------- \\\\\\
 
     /**
-     * Tag to store the barracks pos to NBT.
+     * Our constants. The Schematic names, Defence bonus, and Offence bonus.
      */
-    private static final String TAG_POS = "pos";
+    private static final String SCHEMATIC_NAME = "BarracksTower";
+    private static final int    DEFENCE_BONUS  = 0;
+    private static final int    OFFENCE_BONUS  = 5;
 
     /**
-     * Position of the barracks.
+     * Position of the barracks for this tower.
      */
     private BlockPos barracks = null;
 
     /**
-     * Constructor for the BarracksTower building.
+     * The abstract constructor of the building.
      *
-     * @param c Colony the building is in.
-     * @param l Location of the building.
+     * @param c the colony
+     * @param l the position
      */
-    public BuildingBarracksTower(final Colony c, final BlockPos l)
+    public BuildingBarracksTowerNew(@NotNull final Colony c, final BlockPos l)
     {
         super(c, l);
     }
 
-    /**
-     * Gets the name of the schematic.
-     *
-     * @return Guard schematic name.
-     */
-    @NotNull
+    @Override
+    public int getDefenceBonus()
+    {
+        return DEFENCE_BONUS;
+    }
+
+    @Override
+    public int getOffenceBonus()
+    {
+        return OFFENCE_BONUS;
+    }
+
     @Override
     public String getSchematicName()
     {
-        return GUARD_TOWER;
+        return SCHEMATIC_NAME;
+    }
+
+    @Override
+    public int getMaxBuildingLevel()
+    {
+        return 5;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)